cmd/ical-relay/profiles.go

```diff
@@ -64,20 +66,47 @@ func getProfileCalendar(profile datastore.Profile, profileName string) (*ics.Cal
 		var err error
 
 		for i, s := range profile.Sources {
-			if i == 0 {
-				// first source gets assigned to base calendar
-				log.Debug("Loading source ", s, " as base calendar")
-				calendar, err = getSource(s)
+			sourceCacheFilename := conf.Server.StoragePath + "calstore/" + profileName + "-cache-" + helpers.GetMD5Hash(s) + ".ics"
+			log.Debugf("Loading source calendar %s of profile %s", s, profileName)
+			ncalendar, err = getSource(s)
+			if err != nil {
+				// do not apply cache on base64 as they should always be valid
+				if strings.HasPrefix(s, "base64") {
+					return nil, fmt.Errorf("This should never happen! Error loading base64 source: %s", err.Error())
+				} else if strings.HasPrefix(s, "file") {
+					return nil, fmt.Errorf("This should never happen! Error loading file source: %s", err.Error())
```

Comment:
```suggestion
					return nil, fmt.Errorf("Error loading file source: %s", err.Error())
```

In den angesprochenen Gründen, zB remote file, dead disk oder permission probleme kann der fehler ja tatsächlich auftreten, also würde ich die "This should never happen" warnung nicht nutzen.